src/backend/src/bitcoin_api.rs

```diff
@@ -51,12 +67,61 @@ pub async fn get_all_utxos(
     Ok(all_utxos)
 }
 
-/// Returns the 100 fee percentiles measured in millisatoshi/byte.
-/// Percentiles are computed from the last 10,000 transactions (if available).
-///
-/// Relies on the `bitcoin_get_current_fee_percentiles` endpoint.
-/// See [Bitcoin API](https://internetcomputer.org/docs/current/references/ic-interface-spec/#ic-bitcoin_get_current_fee_percentiles)
-async fn get_current_fee_percentiles(
+/// Sets up periodic refreshing of Bitcoin transaction fee data.
+/// Initializes the cache immediately and configures automatic updates at regular intervals
+pub fn init_fee_percentiles_cache() {
+    ic_cdk::println!(
+        "Initializing fee percentiles cache with {}-second update interval",
+        FEE_PERCENTILES_UPDATE_INTERVAL.as_secs()
+    );
+
+    // Schedule the initial cache population and timer setup to run after init completes
+    set_timer(std::time::Duration::from_secs(0), || {
+        // Set up the recurring timer to update the data
+        set_timer_interval(FEE_PERCENTILES_UPDATE_INTERVAL, || {
+            ic_cdk::spawn(async {
+                let _ = update_fee_percentiles_cache().await;
+            });
+        });
+
+        // Initialize the cache immediately (after init)
+        ic_cdk::spawn(async {
+            let _ = update_fee_percentiles_cache().await;
+        });
+    });
+}
+
+/// Updates fee percentiles for all supported Bitcoin networks (Mainnet, Testnet, Regtest)
+/// in the thread-local cache. Fetches current fee data from the bitcoin canister and stores
+/// it for quick access by other functions.
+async fn update_fee_percentiles_cache() -> Result<(), String> {
+    // Update for each network type
+    for network in &[
+        BitcoinNetwork::Mainnet,
+        BitcoinNetwork::Testnet,
+        BitcoinNetwork::Regtest,
+    ] {
+        match fetch_current_fee_percentiles(*network).await {
```

Comment:
You can dispatch these in parallel, if you wish, with something like:
```
  let results = join_all(networks.map(fetch_current_fee_percentiles));
```
or
```
  let results = join_all(networks.map(|network| {
     match fetch_current_fee_percentiles(network) {
     ....
     }
  }));
```
Not required, though!